go-controller/pkg/ovn/endpoints.go

```diff
@@ -46,7 +46,7 @@ func (ovn *Controller) addEndpoints(ep *kapi.Endpoints) error {
 	for targetPort, ips := range tcpPortMap {
 		for _, svcPort := range svc.Spec.Ports {
 			if svcPort.Protocol == kapi.ProtocolTCP && svcPort.TargetPort.IntVal == targetPort {
-				if svc.Spec.Type == kapi.ServiceTypeNodePort {
+				if svc.Spec.Type == kapi.ServiceTypeNodePort && ovn.NodePortEnable {
```

Comment:
One issue that I see is that NodePortEnable will have to be in sync when called from nodes and the master. Because the service controller runs only on master, while the gateways will run anywhere. Maybe we should use a property on ovn db to mark what the global value is. To avoid inconsistency.

Reply:
I suppose running ovn-k8s-gateway-helper will only work when the central controller is run with nodeport enabled. So that is a clue that configuration is bad. So not sure whether we should optimize it now.

Reply:
We should address it later. Just raising the point.